frontend/src/pages/main-page/MainPage.tsx

```diff
@@ -1,9 +1,29 @@
 import { Typography } from "@mui/material";
+import { http } from "../../providers/http-service";
+import { useQuery } from "@tanstack/react-query";
 
 export default function MainPageContent() {
+  const { isPending, error, data } = useQuery({
```

Comment:
I would recommend extracting this query to a custom hook and just returning all the data you need from it. 

```ts
const { isPending, error, data } = useMainPageData();
```